### PR TITLE
OG-727: remove block number from configuration files in client and server

### PR DESCRIPTION
**Pull request by forshtat**

### CODE REVIEW

File: packages/relay/src/RegistrationManager.ts [hunk 1]
```diff
@@ -61,8 +66,8 @@ export class RegistrationManager {
   txStoreManager: TxStoreManager
   logger: LoggerInterface
 
-  lastMinedRegisterTransaction?: EventData
-  lastWorkerAddedTransaction?: EventData
+  lastMinedRegisterTransaction?: CachedEventData
+  lastWorkerAddedTransaction?: CachedEventData
```
Comment (drortirosh) [inline packages/relay/src/RegistrationManager.ts:70]: I don't see where you need the timestamps of these.

File: packages/relay/src/RegistrationManager.ts [hunk 2]
```diff
@@ -245,17 +259,23 @@ export class RegistrationManager {
   }
 
   _isRegistrationCorrect (): boolean {
-    return isRegistrationValid(this.lastMinedRegisterTransaction, this.config, this.managerAddress)
+    return isRegistrationValid(this.lastMinedRegisterTransaction?.eventData, this.config, this.managerAddress)
   }
 
-  async _queryLatestRegistrationEvent (): Promise<EventData | undefined> {
+  async _queryLatestRegistrationEvent (): Promise<CachedEventData | undefined> {
```
Comment (drortirosh) [inline packages/relay/src/RegistrationManager.ts:265]: maybe a separate PR: if we stop using events , then can use registrar  here too.

File: packages/dev/test/NetworkSimultaion.test.ts [hunk 3]
```diff
@@ -109,7 +109,7 @@ contract('Network Simulation for Relay Server', function (accounts) {
 
       it('should not boost and resend transactions if not that many blocks passed since it was sent', async function () {
         const latestBlock = await env.web3.eth.getBlock('latest')
-        const allBoostedTransactions = await env.relayServer._boostStuckPendingTransactions(latestBlock.number)
+        const allBoostedTransactions = await env.relayServer._boostStuckPendingTransactions(latestBlock.number, parseInt(latestBlock.timestamp.toString()))
```
[the remark this answers is not in the document]
Comment (forshtat) [reply packages/dev/test/NetworkSimultaion.test.ts:112]: I don't understand the second comment. 
Regarding "why not pass latestBlock", the block object is very big and complex. It is also received from the node, which are known to modify the block struct, on a rollup etc. We only need two fields out of it. It is much safer to pass two parameters than entire block object. The code inside this method must not rely on block structure.

File: packages/provider/src/GSNConfigurator.ts [hunk 4]
```diff
@@ -63,12 +61,9 @@ export interface LoggerConfiguration {
  */
 export interface GSNConfig {
   preferredRelays: string[]
-  // number of blocks back the relay will be considered 'active'
-  // must match Relay Server's "activityBlockRate" to be able to discover relay consistently
-  relayLookupWindowBlocks: number
-  // in case access to older logs is restricted, limit number of blocks the client will look for registration info
-  // must match Relay Server's "registrationBlockRate" to be able to discover relay consistently
-  relayRegistrationLookupBlocks: number
+  // in order to avoid pinging relays that are long dead, the client can filter out older registrations
```
[the remark this answers is not in the document]
Comment (forshtat) [reply packages/provider/src/GSNConfigurator.ts:64]: Is there anything that has to be done in this PR for `OG-727`? If no, please create a separate Jira issue.

File: packages/relay/src/RegistrationManager.ts (hunk 1 above)
Comment (forshtat) [reply packages/relay/src/RegistrationManager.ts:70]: As we are now transitioning code to work on timestamps, I made all saved `EventData`s to store the block timestamp as well. It is not currently used for some of those. I can return to the original, but I think keeping the timestamp may end up useful.

File: packages/relay/src/RegistrationManager.ts (hunk 2 above)
Comment (forshtat) [reply packages/relay/src/RegistrationManager.ts:265]: Created `OG-743`

File: packages/common/src/ContractInteractor.ts [hunk 5]
```diff
@@ -1125,15 +1120,20 @@ calculateTransactionMaxPossibleGas: result: ${result}
...
     }
     const relayHub = this.relayHubInstance.address
     if (relayHub == null) {
       throw new Error('RelayHub is not initialized!')
     }
-    const relayInfos = await this.relayRegistrar.readRelayInfos(relayHub, 0, 100)
+    let oldestBlockTimestamp = 0
+    if (relayRegistrationMaximumAge !== 0) {
+      const block = await this.getBlock('latest')
+      oldestBlockTimestamp = Math.max(0, parseInt(block.timestamp.toString()) - relayRegistrationMaximumAge)
```
[the remark this answers is not in the document]
Comment (forshtat) [reply packages/common/src/ContractInteractor.ts]: Fixed. Indeed, this was used a lot :-)

File: packages/dev/test/NetworkSimultaion.test.ts [hunk 6]
```diff
@@ -120,10 +120,10 @@ contract('Network Simulation for Relay Server', function (accounts) {
 
       it('should boost and resend underpriced transactions if the oldest one does not get mined after being sent for a long time', async function () {
         // Increase time by mining necessary amount of blocks
-        await evmMineMany(pendingTransactionTimeoutBlocks)
+        await evmMineMany(pendingTransactionTimeoutSeconds)
```
[the remark this answers is not in the document]
Comment (forshtat) [reply packages/dev/test/NetworkSimultaion.test.ts]: Fixed.

[the remark this answers is not in the document]
Comment (forshtat) [reply packages/dev/test/NetworkSimultaion.test.ts]: Notice: our `increaseTime()` util calls to `evmMine()`

**Review by drortirosh — APPROVED** [review]

**Review by drortirosh — DISMISSED** [review]

**Review by drortirosh — APPROVED** [review]

**Pull request merged**